pkgs/development/compilers/rust/rustc.nix

```diff
@@ -24,6 +24,7 @@
 let
   inherit (lib) optionals optional optionalString concatStringsSep;
   inherit (darwin.apple_sdk.frameworks) Security;
+  useLLVM = stdenv.targetPlatform.useLLVM or false;
```

Comment:
@RossComputerGuy why is this looking at the target platform? I'm possibly missing something but as far as I can see, `useLLVM` is only used to control aspects during the build, not at runtime (these would be covered by #330037) so shouldn't it be `hostPlatform`?

example of a build that fails (when ran on a non-LLVM localSystem):

```
nix-build --arg crossSystem '{ config = "x86_64-unknown-linux-gnu"; useLLVM = true; }' -A buildPackages.rustc.unwrapped
```

(note the `buildPackages`)

Reply:
`pkgsLLVM.buildPackages.stdenv.targetPlatform.useLLVM` evaluates to true while `pkgsLLVM.buildPackages.stdenv.hostPlatform.useLLVM` evaluates to false.

Reply:
yes, that's what I'm saying. in the setup above, because `pkgsLLVM.buildPackages.stdenv.targetPlatform.useLLVM` evaluates to true, the tweaks are enabled when it looks like they shouldn't (because you are building binaries for a platform gcc-based platform so you should be using libgcc rather than libunwind, even if the target is LLVM). is my reasoning correct?